### ISSUE TITLE
Refactor deeply nested if blocks

### ISSUE THREAD

**Issue by vddesai1871** [HTTP-APIs/hydrus#327]
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

Currently the code in `resources.py` has deeply nested if blocks which decrease the readability of the code, also it causes the code to violate PEP max-line-length constraint, trying to avoid that we use line breaks which makes the code even less readable.

### Expected Behaviour:

Instead of using nested Ifs we should check every condition separately and return an appropriate response for that.
Ex.
```
if condition_1:
     if condition_2:
              if condition_3:
                    do_something
              else:
                    response for violation of condition_3
     else
           response for violation of conditon_2
else:
       response for violation of condition_1
```
should become
```
if(!condition_1):
     handle_it
elif(!conditio_2):
     handle_it
elif(!condtion_3):
     handle_it
else:
      do_something
```
So to summarize instead of checking for conditions to be true we should check for false and handle it immediately.

### Do you want to work on this issue?

No

**Comment by realsdx** [comment]
I want to work on this issue. Should i start?

**Comment by Mec-iS** [comment]
#322 (comment)

(These messages will be removed)

**Comment by chrizandr** [comment]
Good idea, although I am unsure if this would help in readability of the code, it is usually easier to understand the `True` cases and then handle failures, than handle failures and do the actual computation in the `else` case. It would certainly reduce the indentation level of it. Rather than this I suggest maybe group similar conditions into a function and handle them at once?

**Comment by Mec-iS** [comment]
Is there any OO pattern we can apply, like Observer on a property or something similar?

**Comment by vddesai1871** [comment]
>Is there any OO pattern we can apply

How about using [Notifications](https://martinfowler.com/eaaDev/Notification.html). As most of the IFs we have are about validation so I think it can be applied.

**Comment by Mec-iS** [comment]
@vddesai1871 How the code would look like? Some dummy example?

**Comment by vddesai1871** [comment]
```
class ValidationError
--attribute error
--function setError
--function getError

function Validate(ValidationError v, other_nessesary_params)
--If any validation step fails set ValidationError.error 
----return false
--else return true
```
Usage
```
v = ValidationError()
if Validate(v,other_params):
   do something important
else
   handle v.error
```

**Comment by chrizandr** [comment]
Wouldn't we have to make classes for each different type of validation then? And again check for each validation in a separate `if` clause? Am I missing something?

**Comment by vddesai1871** [comment]
>Wouldn't we have to make classes for each different type of validation then?

I don't think so, ValidationError.error will have information about which kind of error it is.
>And again check for each validation in a separate if clause? Am I missing something?

Yes, you are correct we will still need multiple if clauses and they will be still checking for negations ( Which as you mentioned is undesirable) just they will be in a separate function.
I, too, would like to avoid that but I can't think of a way to remove this nested ifs without checking for negations(provided how those `check*` functions work). 
A discussion related to the solution I provided earlier https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting

**Comment by kartikpandey596** [comment]
You can use flag to solve this problem.
For example:
Your present condition is:
If (condition):
     If (condition):
          ....
     else:
else:
The solution is:
If(condition):
     If(condition):
           ....   Before reaching the max situation pass check variable value to 1 when if condition is true
Otherwise its default value is 0
    else:
else:
And at last 
Check the condition 
If(check==1):
    If(condition):
         .. Again continue the condition
    else:
else:
In this way you solve this problem

**Comment by Mec-iS** [comment]
As mentioned above, we want to implement a [standard pattern](https://python-patterns.guide/)

**Comment by xadahiya** [comment]
Fixed in #341
